src/Juvix/Nets/Bohm.hs

```diff
@@ -288,43 +326,48 @@ fanInAux2 numFan (numOther, otherLang) level = do
+  let nodeOther1 = RELAuxiliary3 { node       = other1
                                  , primary    = ReLink numFan Aux1
+                                 , auxiliary1 = Link (Port Aux1 fanIn3)
+                                 , auxiliary2 = Link (Port Aux1 fanIn2)
+                                 , auxiliary3 = Link (Port Aux1 fanIn1)
                                  }
-      nodeOther2 = RELAuxiliary0 { node       = other2
+      nodeOther2 = RELAuxiliary3 { node       = other2
                                  , primary    = ReLink numFan Aux2
+                                 , auxiliary1 = Link (Port Aux2 fanIn3)
+                                 , auxiliary2 = Link (Port Aux2 fanIn2)
+                                 , auxiliary3 = Link (Port Aux2 fanIn1)
                                  }
-  traverse_ linkAll [nodeOther1, nodeOther2]
-  deleteRewire [numFan, numOther] [other1, other2]
-
-fanInAux1 ∷ (InfoNetwork net Lang m)
-          ⇒ Node → (Node, Auxiliary1) → Int → m ()
-fanInAux1 numFan (numOther, otherLang) level = do
-  incGraphSizeStep 1
-  other1 ← newNode (Auxiliary1 otherLang)
-  other2 ← newNode (Auxiliary1 otherLang)
-  fanIn1 ← newNode (Auxiliary2 (FanIn level))
-  let nodeOther1 = RELAuxiliary1 { node       = other1
-                                 , primary    = ReLink numFan Aux1
-                                 , auxiliary1 = Link (Port Aux1 fanIn1)
+      nodeFan1   = RELAuxiliary3 { node       = fanIn1
+                                 , primary    = ReLink numOther Aux3
+                                 , auxiliary1 = Link (Port Aux3 other1)
+                                 , auxiliary2 = Link (Port Aux3 other2)
+                                 , auxiliary3 = Link (Port Aux3 other3)
                                  }
-      nodeOther2 = RELAuxiliary1 { node       = other2
-                                 , primary    = ReLink numFan Aux2
-                                 , auxiliary1 = Link (Port Aux2 fanIn1)
+      nodeFan2   = RELAuxiliary3 { node       = fanIn2
+                                 , primary    = ReLink numOther Aux2
+                                 , auxiliary1 = Link (Port Aux2 other1)
+                                 , auxiliary2 = Link (Port Aux2 other2)
+                                 , auxiliary3 = Link (Port Aux2 other3)
                                  }
-      nodeFan1   = RELAuxiliary2 { node       = fanIn1
+      nodeFan3   = RELAuxiliary3 { node       = fanIn3
                                  , primary    = ReLink numOther Aux1
                                  , auxiliary1 = Link (Port Aux1 other1)
                                  , auxiliary2 = Link (Port Aux1 other2)
+                                 , auxiliary3 = Link (Port Aux1 other3)
                                  }
-  traverse_ linkAll [nodeOther1, nodeOther2, nodeFan1]
-  deleteRewire [numFan, numOther] [other1, other2, fanIn1]
+  traverse_ linkAll [nodeOther1, nodeOther2, nodeFan1, nodeFan2, nodeFan3]
```

Comment:
It seems here you forget about `nodeOther3`,and it's linked in the nodeFan's above